wxStaticBox glitch when DPI changes

<!--

IMPORTANT NOTE
--------------

Please note that this entry field uses [Markdown][1], which means that you
need to escape any code snippets pasted here using fenced code blocks,
i.e. by putting ``` before and after it. Also escape, using
backslash, i.e. `\<` any angle brackets to prevent them from being
interpreted as HTML tags.

Use the Preview tab to review your issue before submitting it to verify that
your formatting is correct.

[1]: https://docs.github.com/github/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax

---

Describe the bug:
A clear and concise description of what the bug is.

Expected vs observed behaviour:
Please describe what you expected to happen and what actually happens. Make
sure to explain what exactly is the problem, just attaching a screenshot is
not always sufficient. When in doubt, please provide more details rather
than too few.

Patch or snippet allowing to reproduce the problem:
Please attach the smallest possible patch allowing to reproduce the problem
in a sample. If there is absolutely no appropriate sample, please attach a
minimal self-contained example in a single file.

Skip this step if the problem can be reproduced in one of the samples
without any changes.

To Reproduce:
Steps to reproduce the behaviour, please make them as detailed as possible.
For example:

1. Go to '...'
2. Click on '...'
3. Scroll down to '...'
4. See error
-->

### Description
<!-- Describe the bug here -->

When using a control (e.g. wxStaticBox, wxCheckBox) as the label of a wxStaticBox, it glitches when the DPI changes.

This is an example of a wxStaticBox with wxCheckBox label created on a screen with a DPI scale factor of 1:
![1](https://github.com/wxWidgets/wxWidgets/assets/59702518/de5d5fd8-93ff-4a04-868b-e64fab4901ca)

Moving the same window to a second screen with a DPI scale factor of 1.25 causes it to look like this.
![2](https://github.com/wxWidgets/wxWidgets/assets/59702518/d26e7082-ea47-416a-aeec-b7c841cd4925)

The right of the checkbox label is clipped.
In some scenarios, the text breaks on to a newline where it cannot be seen, and the spacing is broken.

Here is an example of a wxStaticBox with wxStaticText label created on a screen with a DPI scale factor of 1.25. It looks fine.
![4](https://github.com/wxWidgets/wxWidgets/assets/59702518/4d3a4753-2b1f-4c12-94c7-ba34381fcf99)

Contrast this with the same window created on a screen with a DPI scale factor of 1.0 and then moved to a screen with a DPI scale factor of 1.25:
![3](https://github.com/wxWidgets/wxWidgets/assets/59702518/9fc64267-6fc9-44d5-b11f-fe257713071d)
One of the words has moved to the next line and been cut off.
The box around the static box has a large gap.
The spacing under the wxStaticText label is not the same.

I'm using
```
#define wxUSE_DPI_AWARE_MANIFEST 2
```

### Minimum working example
```
#include <wx/wx.h>

class MyFrame:public wxFrame {
public:
    MyFrame(wxWindow* parent);
};

class MyApp:public wxApp {
public:
    bool OnInit() {
        wxFrame* frame=new MyFrame(NULL);
        SetTopWindow(frame);
        frame->Show();
        return true;                    
    }
};

DECLARE_APP(MyApp);
IMPLEMENT_APP(MyApp);

MyFrame::MyFrame(wxWindow* parent):wxFrame(parent,wxID_ANY,_("StatBoxMWE"),wxDefaultPosition,wxDefaultSize,wxDEFAULT_FRAME_STYLE) {
	SetClientSize(wxSize(100,100));
	wxBoxSizer* mainSizer=new wxBoxSizer(wxVERTICAL);
	
	wxCheckBox* cb=new wxCheckBox(this,wxID_ANY,_("checkbox"),wxDefaultPosition,wxDefaultSize,wxALIGN_LEFT);
	wxStaticBox* sb=new wxStaticBox(this,wxID_STATIC,cb);
	wxStaticBoxSizer *sbs=new wxStaticBoxSizer(sb,wxVERTICAL);
	
	wxStaticText* text=new wxStaticText(sb,wxID_ANY,_("hello"),wxDefaultPosition,wxDefaultSize,wxALIGN_LEFT);
	sbs->Add(text,0,wxALL,5);
	
	mainSizer->Add(sbs,1,wxEXPAND|wxALL,10);
	SetSizer(mainSizer);
}
```

### Platform and version information

- wxWidgets version <!-- [e.g. 3.2.1] -->: wxWidgets-3.2.2.1
- wxWidgets port <!-- [e.g. wxMSW, wxGTK, wxOSX] -->: wxMSW
- OS: Windows 10 Enterprise

Hi Vadz
No problem. It's just a minor cosmetic issue under specific circumstances.

I had a look at the solutions you suggested though.

- As far as I can see, there is no virtual `MSWBeforeDPIChangedEvent()` to override for wxStaticBox.

Adding
```
virtual void MSWBeforeDPIChangedEvent(const wxDPIChangedEvent& event) override;
```
to `/include/wx/msw/statbox.h` gives
```
In file included from ../include/wx/statbox.h:70,
                 from ../src/msw/statbox.cpp:25:
../include/wx/msw/statbox.h:79:15: error: 'virtual void wxStaticBox::MSWUpdateOnDPIChange(const wxDP
IChangedEvent&)' marked 'override', but does not override
  virtual void MSWUpdateOnDPIChange(const wxDPIChangedEvent& event) override;
```

- I also tried adding `PositionLabelWindow()` to `wxStaticBox::MSWWindowProc()` but this the application to immediately crash and exit.

I'm probably doing something wrong. I'm not familiar with the inner working of wxwidgets. But I would have been amazed if you (or anyone) had seen straight to the problem without detailed troubleshooting!

--- a/include/wx/msw/statbox.h
+++ b/include/wx/msw/statbox.h
@@ -97,6 +97,8 @@ protected:
 
     void OnPaint(wxPaintEvent& event);
 
+    virtual void MSWBeforeDPIChangedEvent(const wxDPIChangedEvent& event) override;
+
 private:
     void PositionLabelWindow();
 

--- a/src/msw/statbox.cpp
+++ b/src/msw/statbox.cpp
@@ -366,6 +366,11 @@ WXLRESULT wxStaticBox::MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lPar
     return wxControl::MSWWindowProc(nMsg, wParam, lParam);
 }
 
+void wxStaticBox::MSWBeforeDPIChangedEvent(const wxDPIChangedEvent& WXUNUSED(event))
+{
+    PositionLabelWindow();
+}
+
 // ----------------------------------------------------------------------------
 // static box drawing
 // ----------------------------------------------------------------------------

